src/compiler.ts

```diff
@@ -8682,12 +8682,26 @@ export class Compiler extends DiagnosticEmitter {
     var values = expression.values;
     var members = classReference.members;
     var hasErrors = false;
-    var exprs = new Array<ExpressionRef>(numNames + 2);
+    var exprs = new Array<ExpressionRef>();
     var flow = this.currentFlow;
     var tempLocal = isManaged
       ? flow.getAutoreleaseLocal(classReference.type)
       : flow.getTempLocal(classReference.type);
     assert(numNames == values.length);
+
+    // Assume all class fields will be omitted, and add them to our omitted list
+    let omittedClassFieldMembers = new Map<string, string>();
```

Comment:
```suggestion
    let omittedClassFieldMembers = new Set<string>();
```
Let's make this a `Set` and use `_values = Set_values` instead of `_keys = Map_keys` below.